firefox-sync-client: init at 1.7.0

SkamDart opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
A commandline-utility to list/view/edit/delete entries in a firefox-sync account.
Can be used to access bookmarks, passwords, forms, tabs, history, or custom data.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Resolves: https://github.com/NixOS/nixpkgs/issues/311378

pluiedev commented on pkgs/by-name/fi/firefox-sync-client/package.nix:
```diff
@@ -0,0 +1,26 @@
+  vendorHash = "sha256-Gb+4fxMBgvPB9Ki7zIwscY9l2kJ+tuE1Mc3W08YTfk8=";
+
+  meta = with lib; {
+    description = "A commandline-utility to list/view/edit/delete entries in a firefox-sync account.";
```
```suggestion
    description = "Command-line utility to list/view/edit/delete entries in a firefox-sync account";
```

pluiedev commented on pkgs/by-name/fi/firefox-sync-client/package.nix:
```diff
@@ -0,0 +1,26 @@
+    description = "A commandline-utility to list/view/edit/delete entries in a firefox-sync account.";
+    homepage = "https://github.com/Mikescher/firefox-sync-client";
+    license = licenses.asl20;
+    maintainers = [ ];
```
Why not add yourself as a maintainer? We should avoid adding orphaned packages if possible, especially if the package doesn't have an update script

pluiedev commented on pkgs/by-name/fi/firefox-sync-client/package.nix:
```diff
@@ -0,0 +1,26 @@
+  };
+
+  vendorHash = "sha256-Gb+4fxMBgvPB9Ki7zIwscY9l2kJ+tuE1Mc3W08YTfk8=";
+
```
Consider adding an update script
```suggestion
	passthru.updateScript = nix-update-script { };
```

pluiedev reviewed (changes requested):
Accidentally clicked on approve instead of request changes :woman_facepalming:

SkamDart commented:
> Accidentally clicked on approve instead of request changes 🤦‍♀️

No worries. Thank you for reviewing my PR. I went ahead and addressed the comments you had in another commit.

mweinelt commented on maintainers/maintainer-list.nix:
Changes to the maintainer list should go into a separate commit

> maintainers: add skamdart

mweinelt commented on pkgs/by-name/fi/firefox-sync-client/package.nix:
```diff
@@ -0,0 +1,28 @@
+    license = licenses.asl20;
+    maintainers = [ maintainers.skamdart ];
+    mainProgram = "ffsclient";
+    passthru.updateScript = nix-update-script { };
```
One level up, not in `meta`.

mweinelt commented on pkgs/by-name/fi/firefox-sync-client/package.nix:
```diff
@@ -0,0 +1,28 @@
+    maintainers = [ maintainers.skamdart ];
+    mainProgram = "ffsclient";
+  };
+  passthru.updateScript = nix-update-script { };
```
Please between vendorHash and meta, ideally with and empty line before and after. Putting it down here will effectively hide it, as meta is usually the last thing in a package.

SkamDart committed:
maintainers: add skamdart

mweinelt commented:
Did you test the functionality of this package? Would be good to know before we merge it.

SkamDart commented:
@mweinelt I appreciate the thorough review. I believe I have addressed your comments.

SkamDart commented:
> Did you test the functionality of this package? Would be good to know before we merge it.

Recently, there has been an issue with the new user login.
Upstream issue -  https://github.com/Mikescher/firefox-sync-client/issues/13

If you previously used the app and have login saved, then there should not be any issues.
Should we wait until I can validate the new user login flow or update once it is released?

mweinelt commented:
Ideally logging in would be possible with the version we are offering to all users.

Mikescher commented:
> Ideally logging in would be possible with the version we are offering to all users.

fyi: I fixed the login flow in the latest version ([v1.7](https://github.com/Mikescher/firefox-sync-client/releases/tag/v1.7.0))

SkamDart committed:
firefox-sync-client: init at 1.7.0
